node/src/components/consensus/highway_core/active_validator.rs

```diff
@@ -478,82 +483,268 @@ mod tests {
         }
     }
 
-    fn unwrap_single<T: Debug>(vec: Vec<T>) -> T {
-        let mut iter = vec.into_iter();
+    struct TestState {
+        rng: TestRng,
+        state: State<TestContext>,
+        instance_id: u64,
+        fd: FinalityDetector<TestContext>,
+        validators: HashMap<ValidatorIndex, ActiveValidator<TestContext>>,
+        timers: BTreeSet<(Timestamp, ValidatorIndex)>,
+    }
+
+    impl TestState {
+        fn new(
+            state: State<TestContext>,
+            rng: TestRng,
+            start_time: Timestamp,
+            instance_id: u64,
+            fd: FinalityDetector<TestContext>,
+            validators: Vec<ValidatorIndex>,
```

Comment:
Why not just specify the validator count? Or the weights?
Because the indices have to be `0..validator_count` anyway.

Reply:
I wanted to specify explicitly, when `TestState` is constructed, which validators are participating - `ALICE, BOB, CAROL, etc.` rather than using integer, b/c that's what is used to interact with the `TestState` later on and it makes it simpler to understand in my opinion.